### PR TITLE
dosbox-x 0.84.1

### PR DESCRIPTION
**Pull request by Porkepix**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by Porkepix** [comment]
Notes for review:
- mac build scripts got their `x` removed (`build-macosx`-> `build-mac-os);
- While looking at this, I noticed there was also a `build-macos`sdl2`. As we already depended on it, I tried switching to that one;
- If we prefer building against sdl 1.x as it was the case, I guess we can depends on `sdl2` only for Linux then, as sdl 1.x is included in-tree (documentation doesn't seems to show a preference between the both of them);
- This dependency could even be removed as I understand it, it would just use the 1.x version present in-tree;
- Finally a question, why does it depends on a specific version of the `linux-headers`?

**Comment by danielnachun** [comment]
> Notes for review:
> 
> * mac build scripts got their `x` removed (`build-macosx`-> `build-mac-os`);
> * While looking at this, I noticed there was also a `build-macos-sdl2`. As we already depended on it, I tried switching to that one;
> * If we prefer building against sdl 1.x as it was the case and use`build-macos`instead, I guess we can depends on `sdl2` only for Linux then, as sdl 1.x is included in-tree (documentation doesn't seems to show a preference between the both of them or if that's the case I didn't see it);
> * This dependency could even be removed as I understand it, it would just use the 1.x version present in-tree;

We should switch to `sdl2` on macOS if supports it.  `sdl` is deprecated and we are trying to migrate its dependents to `sdl2` whenever possible.

> * Finally a question, why does it depends on a specific version of the `linux-headers`?

Because the Linux headers that we have in CI are too old.  When we switch to 18.04, I think `linux-headers` will be the 4.15 headers and then we'd be able to switch this to just `linux-headers`.

**Comment by Porkepix** [comment]
> > Notes for review:
> > 
> > * mac build scripts got their `x` removed (`build-macosx`-> `build-mac-os`);
> > * While looking at this, I noticed there was also a `build-macos-sdl2`. As we already depended on it, I tried switching to that one;
> > * If we prefer building against sdl 1.x as it was the case and use`build-macos`instead, I guess we can depends on `sdl2` only for Linux then, as sdl 1.x is included in-tree (documentation doesn't seems to show a preference between the both of them or if that's the case I didn't see it);
> > * This dependency could even be removed as I understand it, it would just use the 1.x version present in-tree;
> 
> We should switch to `sdl2` on macOS if supports it. `sdl` is deprecated and we are trying to migrate its dependents to `sdl2` whenever possible.

But for 1.x it doesn't depend on the one provided by Homebrew: 1.x is included in-tree in their repo. That's not the case for 2.x which have to be provided externally (which I did by adding the dependency).

> > * Finally a question, why does it depends on a specific version of the `linux-headers`?
> 
> Because the Linux headers that we have in CI are too old. When we switch to 18.04, I think `linux-headers` will be the 4.15 headers and then we'd be able to switch this to just `linux-headers`.

Do you know when that would be and where is it tracked? `16.04` is already past deprecation (and support on Ubuntu's LTS is a bit of a liar considering only `main` gets the 5 years support, every other repo only gets the standard 9 months). Also, why not jump directly to 20.04/22.04? 18.04 is less than a year away from deprecation.

**Comment by danielnachun** [comment]
> > We should switch to `sdl2` on macOS if supports it. `sdl` is deprecated and we are trying to migrate its dependents to `sdl2` whenever possible.
> 
> But for 1.x it doesn't depend on the one provided by Homebrew: 1.x is included in-tree in their repo. That's not the case for 2.x which have to be provided externally (which I did by adding the dependency).

We should be using brewed dependencies whenever possible anyway, so that's all the more reason for us to use SDL2.

> > Because the Linux headers that we have in CI are too old. When we switch to 18.04, I think `linux-headers` will be the 4.15 headers and then we'd be able to switch this to just `linux-headers`.
> 
> Do you know when that would be and where is it tracked? `16.04` is already past deprecation (and support on Ubuntu's LTS is a bit of a liar considering only `main` gets the 5 years support, every other repo only gets the standard 9 months). Also, why not jump directly to 20.04/22.04? 18.04 is less than a year away from deprecation.

It's being discussed here https://github.com/Homebrew/homebrew-core/issues/105130 and on Slack, and it's a complex discussion.  Right now the major blocker is that we need https://github.com/Homebrew/homebrew-portable-ruby/pull/156 to be finished so that we're using the same CI image for building everything.  There's a few other things that have to be done before the switch but I think we are very close.  Our transition to a newer CI image has been slower because we merge the linuxbrew-core into homebrew-core last year and had to do a lot of cleanup and harmonization between the OSes.  Aside from adding OpenCL support to Linux I consider all the other work from the merger to be done.

**Comment by Porkepix** [comment]
> It's being discussed here #105130 and on Slack, and it's a complex discussion.

I use the occasion to ask, every direct discussions are now taking place on Slack? I used to sometimes go and ask on `#machomebrew` on Freenode, back then. I was looking some times ago for it on libera.chat, but it seems there isn't anything on IRC anymore (aside a few people who just copy and pasted their config and therefore rejoined this chan there), which is a shame imho for a FLOSS project to depends on a closed SaaS service like that :/

**Comment by SMillerDev** [comment]
Slack is only for maintainers, all public discussions are on GitHub.

File: Formula/dosbox-x.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def install
       --disable-dependency-tracking
       --disable-sdltest
     ]
-    build_script = OS.mac? ? "./build-macosx" : "./build"
+    build_script = OS.mac? ? "./build-macos-sdl2" : "./build"
```
Comment (carlocab) [inline Formula/dosbox-x.rb:46]: This doesn't look right, because we don't use `sdl2` on macOS.

**Comment by carlocab** [comment]
> Also, why not jump directly to 20.04/22.04? 18.04 is less than a year away from deprecation.

Yes, it looks like using 18.04 is a non-starter. 22.04 looks like the best option to me.

**Comment by Porkepix** [comment]
> Slack is only for maintainers, all public discussions are on GitHub.

So nothing replaced the old IRC channel as a synchronous communication system? :(
Sad , it could be quite useful.

**Pull request closed without merging**